Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndoHelpers.h

```diff
@@ -23,6 +23,22 @@ namespace AzToolsFramework
             void RemoveLink(
                 TemplateId sourceTemplateId, TemplateId targetTemplateId, const InstanceAlias& instanceAlias, LinkId linkId,
                 PrefabDom linkPatches, UndoSystem::URSequencePoint* undoBatch);
+
+            //! Helper function for adding an entity to a prefab template with undo-redo support.
+            //! @param parentEntityDomBeforeAddingEntity The DOM of the parent entity before adding a new entity under it.
+            //! @param parentEntityDomAfterAddingEntity The DOM of the parent entity after adding a new entity under it.
+            //! @param entityId The id of the entity to be added.
+            //! @param parentEntityId The id of the parent entity under which the new entity is added.
+            //! @param templateId The id of the prefab template under which the new entity DOM will live.
+            //! @param undoBatch The undo batch node to register the add-entity undo nodes to. 
+            void AddEntity(
+                const PrefabDomValue& parentEntityDomBeforeAddingEntity,
+                const PrefabDomValue& parentEntityDomAfterAddingEntity,
+                const PrefabDomValue& newEntityDom,
+                AZ::EntityId entityId,
+                AZ::EntityId parentEntityId,
+                TemplateId templateId,
```

Comment:
nit 1: Can we pass in _AZ::Entity_ and generate entity DOM inside this function?
nit 2: Can this be retrieved from `parentEntityId`?

Reply:
> nit 1: Can we pass in AZ::Entity and generate entity DOM inside this function?

We can. However, PrefabUndoHelpers is a utils file that is supposed to be lightweight. So I don't think making it call other interfaces is a good idea. This is also similar to how other functions in this utils class just ingest doms and forward them.

> nit 2: Can template id be retrieved from parentEntityId?

Yes it can be. However, you'd first need to first fetch its owning instance by looking it up in a map and extract template id from there. Since all of this information is already present in the PrefabPublicHandler, we can avoid additional look up by simply passing along the template id.